app/forms/openid_connect_token_form.rb

```diff
@@ -91,25 +91,35 @@ def pkce_sp
 
   def validate_pkce_or_private_key_jwt
     return if pkce? || private_key_jwt?
-    errors.add :code, t('openid_connect.token.errors.invalid_authentication')
+    errors.add :code,
+               t('openid_connect.token.errors.invalid_authentication'),
+               type: :open_id_connect
   end
 
   def validate_expired
     if identity&.updated_at && identity.updated_at < session_expiration
-      errors.add :code, t('openid_connect.token.errors.expired_code')
+      errors.add :code, t('openid_connect.token.errors.expired_code'), type: :open_id_connect
     end
   end
 
   def validate_code
-    errors.add :code, t('openid_connect.token.errors.invalid_code') if identity.blank? ||
-                                                                       !identity.user
+    if identity.blank? ||
+       !identity.user
```

Comment:
I would bring this back on one line now that it's no longer a trailing if-statement
```suggestion
    if identity.blank? || !identity.user
```